include/alpaka/kernel/TaskKernelCpuOmp2Blocks.hpp

```diff
@@ -92,6 +93,17 @@ namespace alpaka
                 },
                 m_args));
 
+            // Get the OpenMP schedule.
+            auto const schedule(meta::apply(
+                [&](ALPAKA_DECAY_T(TArgs) const&... args) {
+                    return getOmpSchedule<AccCpuOmp2Blocks<TDim, TIdx>>(
+                        m_kernelFnObj,
+                        blockThreadExtent,
+                        threadElemExtent,
+                        args...);
+                },
+                m_args));
```

Comment:
I do not get why we need to pass so many arguments around. All these extents and args are ignored later down the call chain anyway. You could just replace this by: `const auto schedule = GetOmpSchedule<decltype(m_kernelFnObj)>{}();`

What am I missing here?

Reply:
They are for a user (developer of a kernel) if they specialize the trait. E.g. if they want to distinguish between large and small jobs, or smth else. Maybe they even have an internal parameter that defines schedule and so could translate it to alpaka there.

Of course, for alpaka's default logic of just setting some schedule depending on the kernel type only these parameters are not used. But

Reply:
I think from the existing code and commeents it's just not clear yet that there are two ways to specialize the trait and I guess that also didn't help to think about both ways. I will extend.

Reply:
From the example and unit tests it should hopefully be more clear now. Docs TBD

Reply:
Ok, I think I got it now! You want to give users the ability to specialize a trait so they have access to the kernel arguments for determining what kind of schedule to use. I think this is fine then! Thx!